bin/update-ips.bash

```diff
@@ -15,231 +15,351 @@ source "${here}/common.bash"
-    done
+  local yaml_path="${1}"
+  local file="${2}"
+  local IPS
+  shift 2
+  IPS=("$@")
+  tmp_file=$(mktemp --suffix=.yaml)
+
+  yq4 -n '. = []' >"${tmp_file}"
+  for ip in "${IPS[@]}"; do
+    yq4 -i '. |= . + ["'"${ip}"'/32"]' "${tmp_file}"
+  done
 
-    if $DRY_RUN; then
-        out_file=/dev/stdout
-    else
-        out_file=/dev/null
-    fi
-    diff -U3 --color=always \
-        --label "${file//${CK8S_CONFIG_PATH}\//}" <(yq4 -P "${yaml_path}"' // [] | sort_by(.)' "${file}") \
-        --label expected <(yq4 -P '. | sort_by(.)' "${tmp_file}") > "${out_file}"
-    DIFF_RETURN=$?
-    rm "${tmp_file}"
-    return ${DIFF_RETURN}
+  if $DRY_RUN; then
+    out_file=/dev/stdout
+  else
+    out_file=/dev/null
+  fi
+  diff -U3 --color=always \
+    --label "${file//${CK8S_CONFIG_PATH}\//}" <(yq4 -P "${yaml_path}"' // [] | sort_by(.)' "${file}") \
+    --label expected <(yq4 -P '. | sort_by(.)' "${tmp_file}") >"${out_file}"
+  DIFF_RETURN=$?
+  rm "${tmp_file}"
+
+  return ${DIFF_RETURN}
 }
 
 # Fetches the IPs from a specified address
 # Usage: getDNSIPs <dns_record>
 getDNSIPs() {
-    local IPS
-    mapfile -t IPS < <(dig +short "${1}" | grep '^[.0-9]*$')
-    if [ ${#IPS[@]} -eq 0 ]; then
-        log_error "No ips for ${1} was found"
-        exit 1
-    fi
-    echo "${IPS[@]}"
+  local IPS
+  mapfile -t IPS < <(dig +short "${1}" | grep '^[.0-9]*$')
+  if [ ${#IPS[@]} -eq 0 ]; then
+    log_error "No ips for ${1} was found"
+    exit 1
+  fi
+  echo "${IPS[@]}"
 }
 
 # Usage: diffDNSIPs <dns_record> <yaml_path> <file>
 diffDNSIPs() {
-    local IPS
-    read -r -a IPS <<< "$(getDNSIPs "${1}")"
-    diffIPs "${2}" "${3}" "${IPS[@]}"
-    return $?
+  local yaml_path="${2}"
+  local file="${3}"
+
+  diffIPs "${yaml_path}" "${file}" "${@:4}"
+  return $?
 }
 
 # Updates the list from the file and yaml path specified with IPs fetched from the domain
 # Usage: updateDNSIPs <dns_record> <yaml_path> <file>
 updateDNSIPs() {
-    read -r -a IPS <<< "$(getDNSIPs "${1}")"
+  local yaml_path="${2}"
+  local file="${3}"
 
-    yq4 -i "${2}"' = []' "${3}"
-    for ip in "${IPS[@]}"; do
-        yq4 -i "${2}"' |= . + ["'"${ip}"'/32"]' "${3}"
-    done
+  yq4 -i "${yaml_path}"' = []' "${file}"
+
+  IFS=' ' read -ra ip_array <<<"${@:4}"
+
+  for ip in "${ip_array[@]}"; do
+    yq4 -i "${yaml_path}"' |= . + ["'"${ip}"'"]' "${file}"
+  done
 }
 
 # Usage: updateIPs <yaml_path> <file> <IP 1> <IP ..>
 updateIPs() {
-    local yaml_path="${1}"
-    local file="${2}"
-    shift 2
-    local IPS=("$@")
+  local yaml_path="${1}"
+  local file="${2}"
+  shift 2
+  local IPS=("$@")
+  yq4 -i "${yaml_path}"' = []' "${file}"
 
-    yq4 -i "${yaml_path}"' = []' "${file}"
-    for ip in "${IPS[@]}"; do
-        yq4 -i "${yaml_path}"' |= . + ["'"${ip}"'/32"]' "${file}"
-    done
+  IFS=' ' read -ra ip_array <<<"${@:2}"
+
+  for ip in "${ip_array[@]}"; do
+    yq4 -i "${yaml_path}"' |= . + ["'"${ip}"'"]' "${file}"
+  done
 }
```

Comment:
We can change this to just write the whole array at once without clearing and adding each entry individually.